vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -515,16 +515,18 @@ type internal FSharpLanguageService(package : FSharpPackage) =
         
     /// Sync the Roslyn information for the project held in 'projectContext' to match the information given by 'site'.
     /// Also sync the info in ProjectInfoManager if necessary.
-    member this.SyncProject(project: AbstractProject, projectContext: IWorkspaceProjectContext, site: IProjectSite, workspace, forceUpdate, userOpName) =
+    member this.SyncProject(projectContext: IWorkspaceProjectContext, site: IProjectSite, workspace: VisualStudioWorkspaceImpl, forceUpdate, userOpName) =
         let wellFormedFilePathSetIgnoreCase (paths: seq<string>) =
             HashSet(paths |> Seq.filter isPathWellFormed |> Seq.map (fun s -> try Path.GetFullPath(s) with _ -> s), StringComparer.OrdinalIgnoreCase)
 
         let mutable updated = forceUpdate
 
+        let project = workspace.CurrentSolution.Projects |> Seq.filter (fun p -> p.Name = projectContext.DisplayName) |> Seq.exactlyOne
```

Comment:
I'm implicitly assuming here that there isn't any batching happening with F# projects. Since this seems to happen after OnSolutionLoaded is triggered (I think?) that would be a safe assumption, since the only batching you'd get would be that batching.